ElectronicObserver/Window/Tools/SortieRecordViewer/Sortie/Battle/Phase/PhaseNightBattle.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using ElectronicObserver.Data;
+using ElectronicObserver.Properties.Data;
+using ElectronicObserverTypes;
+using ElectronicObserverTypes.Attacks;
+
+namespace ElectronicObserver.Window.Tools.SortieRecordViewer.Sortie.Battle.Phase;
+
+public class PhaseNightBattle : PhaseBase
+{
+	public override string Title => BattleRes.BattlePhaseNightBattle;
+
+	private List<PhaseNightBattleAttack> Attacks { get; } = new();
+	public List<PhaseNightBattleAttackViewModel> AttackDisplays { get; } = new();
+
+	public PhaseNightBattle(List<FleetFlag> apiAtEflag, List<int> apiAtList,
+		List<List<HitType>> apiClList, List<List<double>> apiDamage, List<List<int>> apiDfList,
+		List<int> apiNMotherList, List<List<object>> apiSiList, List<NightAttackKind> apiSpList)
+	{
+		static int ParseInt(object e) => e switch
+		{
+			JsonElement { ValueKind: JsonValueKind.Number } n => n.GetInt32(),
+			JsonElement { ValueKind: JsonValueKind.String } s => int.Parse(s.GetString()!),
+			_ => throw new NotImplementedException(),
+		};
+
+		List<FleetFlag> fleetflag = apiAtEflag;
```

Comment:
Is there a reason to assign those parameters to new variables ?

Reply:
Not really, but I'd prefer to keep it like this for now to match the old code as close as possible.